### PR TITLE
Minor cleanup in the VTU writer.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
```
Jenkins: serial
===============
The following tests FAILED:
	1545 - data_out/data_out_base_vtu_02.debug (Failed)
	2524 - grid/grid_out_08.debug (Failed)

Jenkins: MPI
============
The following tests FAILED:
	4263 - mpi/codim_01.mpirun=3.debug (Failed)
```

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -5811,23 +5811,17 @@ namespace DataOutBase
...
-            node_coordinates_3d.emplace_back(node_position[1]);
-          else
-            node_coordinates_3d.emplace_back(0.0f);
-
-          if (spacedim >= 3)
-            node_coordinates_3d.emplace_back(node_position[2]);
-          else
-            node_coordinates_3d.emplace_back(0.0f);
+          for (unsigned int d = 0; d < 3; ++d)
+            if (d < dim)
```
Comment (marcfehling) [inline source/base/data_out_base.cc:5821]:
```suggestion
            if (d < spacedim)
```

**Review by marcfehling — APPROVED** [review]
I think this should fix it.

File: source/base/data_out_base.cc (hunk 1 above)
Comment (bangerth) [reply source/base/data_out_base.cc]: Ugh, yes. Let's see if this works now!

**Comment by marcfehling** [comment]
Now a different test fails?!
```
The following tests FAILED:
	4826 - mpi/petsc_step-27.mpirun=2.debug (Failed)
```

**Comment by bangerth** [comment]
I rebased on the test now seems to succeed. The test that failed doesn't actually use `DataOut`, so the failure must have had other, spurious causes.

**Pull request merged**